### ISSUE TITLE
Critical lack of {total} pseudomods

### ISSUE THREAD

**Issue by LongJohnDude** [Kyusung4698/PoE-Overlay#230]
The problem is that many modes, due to their variability, but at the same time the same essence, will be useless in a search without pseudo-modes namely and most importantly {total} pseudomod.

For example, I have a dagger, with a crafted crit chance for spells, say something like 50% global crit implicit and crafted 30%. At the moment, the search works in such a way that I see in the results only something really similar and it will have exact **crafted** spellcrit mod only. And no manipulations in the item evaluation window will allow me to get into the search result for example:
- 30% global + 50% crafted
- 40% global + 40% natural roll
- corrupted in anything implicit + 80% natural roll
... etc.

The same situation will be true for spell damage, cast speed, elemental damage, global crit multi, elemental penetration and so on. That is, this is true for any item base type that has a non-unique implicit that may be present in some form in its rolls/mods.

Ok, in **some** cases we can skip implicit value coz search can to take into account item class (but not base type, this leads to the opposite - unnecessary narrowing of search results) and just increase the value of the corresponding suffix/prefix (if it is of course available on this item). But this will not work if it is not pseudo or simply absent. In any case this is a crutch.

So I say that **all global item mods must be use pseudo {total}** otherwise the search becomes, to put it mildly, extremely inaccurate. This problem is mostly unique to weapons. 

For other slots it is either not so critical, with the exception of what I would say Opal/Still rings and some Talismans and some corrupts.  
Nevertheless, it is present in Marble/Paua/Blue Pearl/Gold Amulets, some Talismans, Rustic Sash, Cloth/Studded/Vanguard Belts, Diamond/Gold/Iron/Vermillion/Cerulean Rings, Shields with Spell Damage/Dodge Spell/Attack Hits implicit, all Quivers (hope I haven’t missed anything).
Or it falls into the category {total} (like Two-Toned Boots will give total res) or you anyway will use this unique implicit/base type to search (like Bone Helmet).

It is necessary that they are either considered as {total} by default, or we had a simple opportunity to change them to {total} in a couple of clicks in the search coz separation to implicit/natural/crafted is not right way for search in most cases.

**Comment by Kyusung4698** [comment]
## 0.5.16 (2020-02-13)

- add additional pseudo mods (#230)

Let me know if there are still some mods missing you require. Thanks for your feedback!

**Comment by LongJohnDude** [comment]
This is a great change!
So, I conducted a fairly large test (unfortunately I could not do it right after the patch, due to the lack of free time). 

There are some {total} mods on items where this {total} is useless, because there can be no different mods that can be summed up, but redundancy is better. especially in the future, the mod pool may change and you will have to redo this part of the program again. This is more polishing than ensuring proper performance.
In general, everything works fine and I will simply list below those cases when something went wrong.

1. 
```
Rarity: Rare
Kraken Barb
Cobalt Jewel
--------
Item Level: 85
--------
5% increased Attack Speed while Dual Wielding
6% increased Attack Speed with Wands
+15% to Lightning Resistance
3% increased Attack and Cast Speed
--------
Place into an allocated Jewel Socket on the Passive Skill Tree. Right click to remove from the Socket
```

![image](https://user-images.githubusercontent.com/34474044/74606728-805f2380-50db-11ea-999a-8c2f68da45ef.png)

"Attack and Cast Speed" mod, in search it separates on two "total Attack Speed" and "total Cast Speed" (and in the menu there is no choice of this particular mod) and for some reason tradesite don't return in result any hybrid "Attack and Cast Speed" if you select them one or both.

2.
```
Rarity: Rare
Sol Gorget
Citrine Amulet
--------
Requirements:
Level: 64
--------
Item Level: 85
--------
+23 to Strength and Dexterity (implicit)
--------
+29 to Strength
19% increased Lightning Damage
19% increased Evasion Rating
+76 to maximum Life
Regenerate 2.5 Life per second
38% increased Elemental Damage with Attack Skills
```

![image](https://user-images.githubusercontent.com/34474044/74606554-185c0d80-50da-11ea-98bc-c0451fa753d4.png)

{total} "increased Lightning Damage with Attack Skills" don't sum "increased Lightning Damage" and "increased Elemental Damage with Attack Skills" (for example similar "increased Spell Damage" and "increased Fire Damage" sum correctly to "increased Fire Spell Damage"). 

3. Implicit (Adds X to Y Physical Damage to Attacks) on Steel/Iron rings and Serrated/Broadhead Arrow Quiver don't have {total} with (Adds X to Y Physical Damage to Attacks) prefix mod.

4. BTW IMO you can sum dagger's implicit "increased Global Critical Strike Chance" with natural/crafted "increased Critical Strike Chance for Spells", In any case, the tradesite works correctly with such a sum if you use "+#% total Critical Strike Chance for Spells"

P.S. If I find more, I will add a comment in this issue. 

P.P.S. This not relevant to this issue, but I would like to mention this - except for the problem of normalizing the quality of armors and weapons, during assessment does not apply search range penalty to final amount of armour/evasion/energy shield and DPS/pDPS/eDPS, attack speed, critical strike chance. Perhaps this is correct decision, but at least we should be able to change these values in the item assessment menu like all other one mods.

**Comment by Kyusung4698** [comment]
> ![image](https://user-images.githubusercontent.com/34474044/74606728-805f2380-50db-11ea-999a-8c2f68da45ef.png)
> 
> "Attack and Cast Speed" mod, in search it separates on two "total Attack Speed" and "total Cast Speed" (and in the menu there is no choice of this particular mod) and for some reason tradesite don't return in result any hybrid "Attack and Cast Speed" if you select them one or both.
> 

I'll remove the 'attack_and_cast_speed_+%' modifier from this group. Seems like the tradesite does not add those. See:

![image](https://user-images.githubusercontent.com/38520678/74644471-9e2d9680-5176-11ea-850a-2b5cf2099110.png)

> {total} "increased Lightning Damage with Attack Skills" don't sum "increased Lightning Damage" and "increased Elemental Damage with Attack Skills" (for example similar "increased Spell Damage" and "increased Fire Damage" sum correctly to "increased Fire Spell Damage").

Will be added!

> 1. Implicit (Adds X to Y Physical Damage to Attacks) on Steel/Iron rings and Serrated/Broadhead Arrow Quiver don't have {total} with (Adds X to Y Physical Damage to Attacks) prefix mod.

There was a missmatch in my stats data which caused this implicit mod not beeing recognized as "Adds # to # ...". Will be fixed.

> 2. BTW IMO you can sum dagger's implicit "increased Global Critical Strike Chance" with natural/crafted "increased Critical Strike Chance for Spells", In any case, the tradesite works correctly with such a sum if you use "+#% total Critical Strike Chance for Spells"

Will be changed! 

> P.P.S. This not relevant to this issue, but I would like to mention this - except for the problem of normalizing the quality of armors and weapons, during assessment does not apply search range penalty to final amount of armour/evasion/energy shield and DPS/pDPS/eDPS, attack speed, critical strike chance. Perhaps this is correct decision, but at least we should be able to change these values in the item assessment menu like all other one mods.

This is more a historical issue. The mod ranges were introduced after the property section have been implemented. Will be added.

Thank you for your great input! 🚀 👍

**Comment by Kyusung4698** [comment]
## 0.5.18 (2020-02-19)
- update pseudo config (#230)
    - add global critical strike chance to critical strike chance for spells
    - add increased elemental damage to attack skills
    - remove hybrid attack/cast speed from total attack/cast speed

**Comment by Kyusung4698** [comment]
Closed in favour of #294
